### PR TITLE
Fix UC secret update and correct syntax of a gcp test

### PR DESCRIPTION
**Pull request by shreyas-goenka**
Makes `recipient_profile_str` a write only field thus fixing the ACC test

### CODE REVIEW

File: catalog/acceptance/provider_test.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ func TestUcAccCreateProviderDb2Open(t *testing.T) {
 			  recipient_profile_str = jsonencode({
 					"shareCredentialsVersion":1,
 					"bearerToken":"dapiabcdefghijklmonpqrstuvwxyz",
-					"endpoint":"https://sharing.delta.io/delta-sharing/"}
+					"endpoint":"https://sharing.delta.io/delta-sharing/"
```
Comment (shreyas-goenka) [inline catalog/acceptance/provider_test.go:22]: Fixes the syntax issue, but the gcp test is still broken

File: catalog/resource_provider.go [hunk 2]
```diff
@@ -51,15 +57,19 @@ func (a ProvidersAPI) updateProvider(ci *ProviderInfo) error {
 }
 
 func ResourceProvider() *schema.Resource {
-	providerSchema := common.StructToSchema(ProviderInfo{}, func(m map[string]*schema.Schema) map[string]*schema.Schema {
+	providerInfoSchema := common.StructToSchema(ProviderInfo{}, func(m map[string]*schema.Schema) map[string]*schema.Schema {
+		m["authentication_type"].ValidateFunc = validation.StringInSlice([]string{"TOKEN"}, false)
```
Comment (nfx) [inline catalog/resource_provider.go:61]: and where `recipient_profile_str` will be coming from?...

**Review by nfx — CHANGES_REQUESTED** [review]

File: catalog/resource_provider.go (hunk 2 above)
Comment (nfx) [reply catalog/resource_provider.go:61]: you need to write `m["recipient_profile_str"].DiffSuppressFunc`, which will suppress the diff when platform returns zero value and we have some in the config. this is "inverted" behavior from `tf:"suppress_diff"` and has to be hand-coded

File: catalog/resource_provider.go [hunk 3]
```diff
@@ -51,15 +57,19 @@ func (a ProvidersAPI) updateProvider(ci *ProviderInfo) error {
 }
 
 func ResourceProvider() *schema.Resource {
-	providerSchema := common.StructToSchema(ProviderInfo{}, func(m map[string]*schema.Schema) map[string]*schema.Schema {
```
Comment (nfx) [inline catalog/resource_provider.go:54]: here's how we can make this more readable and clear:

1. revert the `providerSchema` rename. It diverts the attention from the intention of this commit by making more changes than necessary.
2. remove `ProviderInfoMetadata` as it's not necessary anymore.
3. create the copy of `providerSchema`, but without `recipient_profile_str`:

```go
providerSchemaForRead := map[string]*schema.Schema{
  "name": providerSchema["name"],
  "comment": providerSchema["comment"],
  "authentication_type": providerSchema["authentication_type"],
}
```

you can also just inline that in the `Read` callback.

**Review by nfx — CHANGES_REQUESTED** [review]

File: catalog/resource_provider.go (hunk 3 above)
Comment (shreyas-goenka) [reply catalog/resource_provider.go]: Done

**Comment by shreyas-goenka** [comment]
test run for this PR: https://github.com/databricks/eng-dev-ecosystem/actions/runs/3801210820

**Review by nfx — APPROVED** [review]

**Pull request merged**